source/agora/utils/Log.d

```diff
@@ -198,6 +206,49 @@ public class CircularAppender : Appender
     }
 }
 
+///
+unittest
+{
+    class MockLayout : Appender.Layout
+    {
+        /// Format the message
+        public override void format (LogEvent event, scope FormatterSink dg)
+        {
+            sformat(dg, "{}", event);
+        }
+    }
+
+    auto get_log_output = (const string log_msg){
+        import ocean.util.log.ILogger;
+
+        immutable buffer_size = 7;
+        char[buffer_size + 1] result_buff;
+        LogEvent event;
+        auto appender = getCircularAppender!(buffer_size)();
+        appender.clear();
+        event.set(ILogger.Context.init, ILogger.Level.init, log_msg, "");
+        appender.layout(new MockLayout());
+        appender.append(event);
+        appender.print(result_buff[]);0 .. min(log_msg.length, buffer_size)]);
+        return result_buff[0 .. min(log_msg.length, buffer_size)].dup;
+    };
+
+    // case 1
+    // the internal buffer size is greater than the length of the messages 
+    // that we are trying to log
+    assert(get_log_output("01234") == "01234");
+
+    // case 2
+    // the internal buffer size is equal to the length of the messages 
+    // that we are trying to log(there is a terminating newline)
+    assert(get_log_output("012345") == "012345");
```

Comment:
Shouldn't this be:
```suggestion
    assert(get_log_output("012345") == "012345\n");
```
?

Reply:
get_log_output doesn't return the newline, let me know if you want me to change this

Reply:
Missed that. Works for me.